.github/workflows/push.yml

```diff
@@ -0,0 +1,16 @@
+name: Build on push
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    uses: ./.github/workflows/build-yocto.yml
+  test:
+    uses: ./.github/workflows/test.yml
```

Comment:
you need to check that the build completed with success here too, before running the tests